firmware-support/bittide-sys/src/axi.rs

```diff
@@ -116,42 +117,46 @@ impl AxiRx {
 
 #[derive(uDebug, Copy, Clone)]
 pub struct AxiTx {
-    payload_addr: *mut u8,
+    base_addr: *mut u8,
 }
 
 impl AxiTx {
-    pub fn new(payload_addr: *mut u8) -> Self {
-        AxiTx { payload_addr }
+    /// Creates a new instance of `AxiTx`.
+    /// # Safety
+    /// - `base_addr` Must be the base address of the Axi Tx peripheral.
+    pub unsafe fn new(base_addr: *const ()) -> Self {
+        AxiTx {
+            base_addr: base_addr as *mut u8,
+        }
     }
 
     pub fn send(&mut self, packet: &[u8]) {
         // Split packet into a slice of words and a slice of bytes
         let len = packet.len();
         let words = len / 4;
-        let (words_slice, bytes_slice) = packet.split_at(words * 4);
+        let (init_slice, bytes_slice) = packet.split_at(words * 4);
 
+        // Create a slice of u32 from first multiple of 4 bytes
+        let words_slice =
+            unsafe { core::slice::from_raw_parts(init_slice.as_ptr() as *const u32, words) };
```

Comment:
This is not safe, `packet` can be any arbitrary slice, so it doesn't have any alignment guarantees, but for a `&[u32]` it needs to be aligned to 4 bytes. So either this whole function must be marked as `unsafe` and this constraint must be added to the `# Safety` section, or the alignment needs to be checked at runtime to make sure constructing a `&[32]` from the `&[u8]` is safe to do.

Maybe you can use [`.align_to()`](https://doc.rust-lang.org/stable/core/primitive.slice.html#method.align_to) here?

Reply:
Could I instead check for aligned with https://doc.rust-lang.org/std/primitive.pointer.html#method.is_aligned and otherwise panic? I'm afraid this might introduce significant overhead which I'd like to avoid on embedded platforms.